src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -162,13 +168,16 @@ RuntimeException potentiallyConvertRuntimeException(RuntimeException ex) {
 	@Override
 	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
 		prepareIndexCreator(applicationContext);
+		templateSupport.setApplicationContext(applicationContext);
```

Comment:
Why do we need to set the application context und support and reactive like this, can't we make both application context aware itself and let it be injected by spring?

Reply:
CouchbaseTemplateSupport isn't a bean, it isn't create or managed by Spring.

CouchbaseTemplate (and ReactiveCouchbaseTemplate) are beans. They are created by Spring, and since they are ApplicationContextAware, Spring calls their setApplicationContext() method after creating them.

During the construction of CouchbaseTemplate (and ReactiveCouchbaseTemplate), the CouchbaseTemplateSupport object is created like this :

	this.templateSupport = new CouchbaseTemplateSupport(converter);

CouchbaseTemplate (and Reactive) do not have the application context at that point. They only get the applicationcontext afterwards, when their setApplicationContext() method is called.  And that's a good place to propagate the application context to the CouchbaseTemplateSupport object.